src/middlewared/middlewared/service/decorators.py

```diff
@@ -25,6 +25,19 @@ def filterable(fn):
     return accepts(Ref('query-filters'), Ref('query-options'))(fn)
 
 
+# When we want to specify roles with filterable call this
+# i.e. @filterable -> @filterable_with_roles(['role1',...])
+# MUCH simplier than the alternatives.
```

Comment:
IMO you write the implementation once, but you use it multiple times, so the simplicity of the usage shall be preferred over the simplicity of implementation. E.g. built-in `@dataclass` decorator can be used both as `@dataclass` and `@dataclass()`, that's what we can use as an example.

Reply:
Good point, thanks.  I'll address.